File: frontend/other/0_python/tasks/normal/20.py
'''
Удаление дубликатов из списка
Ввод: список
Вывод: новый список без повторяющихся значений, сохраняя порядок
'''
import unittest
def remove_duplicates(lst):
  seen = set()
  result = []
  for item in lst:
    if item not in seen:
      seen.add(item)
      result.append(item)
  return result
class TestRemoveDuplicates(unittest.TestCase):
  def test_table(self):
    test_cases = [
      ([1, 2, 2, 3, 1, 4], [1, 2, 3, 4]),
      (['a', 'b', 'a', 'c'], ['a', 'b', 'c']),
      ([], []),
      ([1, 1, 1], [1]),
      ([42], [42])
    ]
    for input_data, expected_output in test_cases:
      with self.subTest(input=input_data):
        self.assertEqual(remove_duplicates(input_data), expected_output)
unittest.main()

File: frontend/4_js/other/tasks/normal/20.js
/**
 * Удаление дубликатов из списка
 * Ввод: список
 * Вывод: новый список без повторяющихся значений, сохраняя порядок
 */
function removeDublicates(arr) {
  const result = [];
  const seen = new Set();
  for (let item of arr) {
    if (!seen.has(item)) {
      seen.add(item);
      result.push(item);
    }
  }
  return result;
}
console.log(removeDublicates([1, 2, 2, 3, 1, 4]));
